### PR TITLE
[AC+SMP] deprecating upstream_proxy in admission control

### PR DESCRIPTION
**Pull request by christinacelee**
## Summary

With PR #2125 , client transaction submissions are handled by shared mempool, so `AdmissionControl::upstream_proxy` is deprecated. This PR cleans up this deprecated code and refactors as necessary. 

Notes:
- Prometheus counters for `admission_control` that are removed in this PR will be added back to shared mempool in a future PR
- I had to increase the type to u64 to u128 in `core_mempool::get_required_balance` because of multiplication overflow
- Added mock shared mempool for admission-control-fuzzing

## Test Plan

Existing/refactored tests pass

### CODE REVIEW

File: admission_control/admission-control-service/src/runtime.rs [hunk 1]
```diff
@@ -1,39 +1,28 @@
...
-use network::validator_network::{AdmissionControlNetworkEvents, AdmissionControlNetworkSender};
-use std::{collections::HashMap, net::ToSocketAddrs, sync::Arc};
+use libra_config::config::NodeConfig;
+use std::{net::ToSocketAddrs, sync::Arc};
 use storage_client::{StorageRead, StorageReadServiceClient};
 use tokio::runtime::{Builder, Runtime};
-use vm_validator::vm_validator::VMValidator;
 
 /// Handle for AdmissionControl Runtime
 pub struct AdmissionControlRuntime {
```
Comment (phoenix-antigravity) [inline admission_control/admission-control-service/src/runtime.rs]: there's no point to have Runtime anymore
it's just a wrapper around AC service
can we get rid of it?

File: libra-node/src/main_node.rs [hunk 2]
```diff
@@ -213,14 +211,7 @@ pub fn setup_environment(node_config: &mut NodeConfig) -> LibraHandle {
             .add_mempool(vec![ProtocolId::from_static(MEMPOOL_DIRECT_SEND_PROTOCOL)]);
         mempool_network_handles.push((network.peer_id, mempool_sender, mempool_events));
 
-        let (ac_sender, ac_events) =
-            network_provider.add_admission_control(vec![ProtocolId::from_static(
-                ADMISSION_CONTROL_RPC_PROTOCOL,
```
Comment (phoenix-antigravity) [inline libra-node/src/main_node.rs]: can we also get rid of all references in networking code to  AC rpc protocol

File: mempool/src/lib.rs [hunk 3]
```diff
@@ -2,7 +2,6 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #![forbid(unsafe_code)]
-#![deny(missing_docs)]
```
Comment (phoenix-antigravity) [inline mempool/src/lib.rs]: why we had to remove this?

File: admission_control/admission-control-service/src/runtime.rs (hunk 1 above)
Comment (christinacelee) [reply admission_control/admission-control-service/src/runtime.rs]: `tokio::runtime::Runtime` is not cloneable, so if we want to keep `AdmissionControlService` cloneable we need `AdmissionControlRuntime` to serve as a handle to the runtime

File: mempool/src/lib.rs (hunk 3 above)
Comment (sunmilee) [reply mempool/src/lib.rs]: Just add in documentation/comments and you won't get clippy errors

File: mempool/src/shared_mempool.rs [hunk 4]
```diff
@@ -476,7 +476,7 @@ async fn gc_task(mempool: Arc<Mutex<CoreMempool>>, gc_interval_ms: u64) {
 ///   - outbound_sync_task (task that periodically broadcasts transactions to peers)
 ///   - inbound_network_task (task that handles inbound mempool messages and network events)
 ///   - gc_task (task that performs GC of all expired transactions by SystemTTL)
-pub(crate) fn start_shared_mempool<V>(
```
Comment (sunmilee) [inline mempool/src/shared_mempool.rs]: Does this need to be public just for  fuzzing?

File: libra-node/src/main_node.rs (hunk 2 above)
Comment (christinacelee) [reply libra-node/src/main_node.rs]: I can remove `ADMISSION_CONTROL_RPC_PROTOCOL` refs in this file for this PR - I can delete the actual declaration of it, along with AdmissionControlNetworkEvents and AdmissionControlNetworkSender, in another PR - this PR is getting too big

**Pull request closed without merging**